From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Wed, 29 Dec 2021 15:12:40 +1000
Subject: [PATCH] Ensure projectileSource is loaded from the ownerUUID


diff --git a/src/main/java/net/minecraft/world/entity/projectile/Projectile.java b/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
index d827fb1768c7f3ecbb65b9477aac6794253e01b7..689c2be4a6fdb4f54b44914780c09cc1a166e6d0 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/Projectile.java
@@ -113,6 +113,8 @@ public abstract class Projectile extends Entity implements TraceableEntity {
             }
             // Paper end
             this.refreshProjectileSource(false); // Paper
+            this.projectileSource = null; // MultiPaper - set the projectileSource
+            if (this.cachedOwner != null && !this.cachedOwner.isRemoved()) this.projectileSource = (ProjectileSource) cachedOwner.getBukkitEntity(); // MultiPaper - set the projectileSource
             return this.cachedOwner;
         } else {
             return null;
