From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PureGero <puregero@gmail.com>
Date: Fri, 21 Jan 2022 09:58:45 +1000
Subject: [PATCH] Reset mustNotSave on chunk load


diff --git a/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java b/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
index 3d2324759bc154d369fa6d84671397d244802f3c..cae24e0cd477df6d83561e0d300038f8c31b53b8 100644
--- a/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
@@ -896,6 +896,7 @@ public class LevelChunk extends ChunkAccess {
     // CraftBukkit start
     public void loadCallback() {
         if (this.loadedTicketLevel) { LOGGER.error("Double calling chunk load!", new Throwable()); } // Paper
+        this.mustNotSave = false; // MultiPaper - Reset mustNotSave on chunk load
         // Paper - rewrite chunk system - move into separate callback
         this.loadedTicketLevel = true;
         // Paper - rewrite chunk system - move into separate callback
