From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xymb <xymb@endcrystal.me>
Date: Sun, 29 Jan 2023 16:33:09 +0100
Subject: [PATCH] Fix Ikea V1 dupe


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index e92f9c3a44e0c7b4a5ca748b6aaf28defe9d6b02..73207eb468b1e5a8eb2be490206378a058cdde60 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -3175,6 +3175,11 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
                     return entity1 != entity;
                 }).collect(ImmutableList.toImmutableList());
             }
+            // MultiPaper start
+            if (entity instanceof ServerPlayer player) {
+                player.closeContainer();
+            }
+            // MultiPaper end
 
             entity.boardingCooldown = 60;
             this.gameEvent(GameEvent.ENTITY_DISMOUNT, entity);
